crates/workspace/src/workspace.rs

```diff
@@ -941,6 +941,25 @@ impl Workspace {
                     );
                 }
 
+                project::Event::LanguageServerShowDocument(request) => {
+                    if request.external {
```

Comment:
Handle the request only if the server requests that we open it in an external program - https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#showDocumentParams

I'm unclear as to how Zed should handle the request for a non-external request.